### PR TITLE
Sync from apache/kafka (8 August 2019) and upgrade avro to 1.9.0

### PR DESCRIPTION
**Pull request by ijuma**
The avro upgrade was needed to fix the following error during
':support-metrics-client:generateAvro':

> Caused by: org.apache.velocity.exception.MethodInvocationException: Variable $velocityCount has not been set at ...

Conflicts:
 * gradle.properties -> trivial fix, `scalaVersion` is next to `version`, which
 is different in ccs kafka
 * gradle/dependencies.gradle -> removed unused `httpclient` and `httpmime` in
 the `versions` array since they were unused and were the reason for the
 divergence with upstream.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -65,19 +65,18 @@ versions += [
...
   avroPlugin: "0.10.0",
-  bcpkix: "1.61",
+  bcpkix: "1.62",
   checkstyle: "8.20",
   commonsCli: "1.4",
   gradle: "5.4.1",
   gradleVersionsPlugin: "0.21.0",
   grgit: "3.1.1",
-  httpcomponents: "4.5.7",
-  httpclient: "4.5.8",
```
Comment (stanislavkozlovski) [inline gradle/dependencies.gradle]: `httpclient` is preent in AK/trunk - https://github.com/apache/kafka/blob/66d81a0e50680916f541703049dd4e911542b5ed/gradle/dependencies.gradle#L72

Best remove it from there if really is unused?

File: gradle/dependencies.gradle [hunk 2]
```diff
@@ -65,19 +65,18 @@ versions += [
...
-  httpmime: "4.5.2",
+  httpcomponents: "4.5.9",
```
Comment (stanislavkozlovski) [inline gradle/dependencies.gradle]: This `httpcomponents` is unused and not in AK.

File: gradle/dependencies.gradle (hunk 1 above)
Comment (ijuma) [reply gradle/dependencies.gradle]: It's used there, but not here. I'll reduce the divergence further.

File: gradle/dependencies.gradle (hunk 2 above)
Comment (ijuma) [reply gradle/dependencies.gradle]: It's used here, but not in AK. I'll reduce the divergence.

**Comment by ijuma** [comment]
Flaky tests that failed:

> kafka.admin.ReassignPartitionsClusterTest.testHwAfterPartitionReassignment
> kafka.api.AdminClientIntegrationTest.testIncrementalAlterConfigsForLog4jLogLevels
> kafka.api.AdminClientIntegrationTest.testIncrementalAlterConfigsForLog4jLogLevelsCanResetLoggerToCurrentRoot
> kafka.api.PlaintextConsumerTest.testFetchInvalidOffset
> kafka.api.SaslGssapiSslEndToEndAuthorizationTest.testNoDescribeProduceOrConsumeWithoutTopicDescribeAcl
> kafka.api.SaslPlainSslEndToEndAuthorizationTest.testNoDescribeProduceOrConsumeWithoutTopicDescribeAcl

Restarted the build.

**Comment by ijuma** [comment]
More flaky failures (subset of the previous ones):

> kafka.api.AdminClientIntegrationTest.testIncrementalAlterConfigsForLog4jLogLevels
> kafka.api.AdminClientIntegrationTest.testIncrementalAlterConfigsForLog4jLogLevelsCanResetLoggerToCurrentRoot
> kafka.api.DelegationTokenEndToEndAuthorizationTest.testNoDescribeProduceOrConsumeWithoutTopicDescribeAcl

@stanislavkozlovski has a PR for the log4j log levels issue. There are JIRAs in Apache Kafka for the other flaky tests.

**Comment by ijuma** [comment]
Merging to `master`.

**Pull request merged**